unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneController.cs

```diff
@@ -43,6 +43,9 @@ public void Initialize()
 
             PoolManager.i.OnGet -= Environment.i.platform.cullingController.objectsTracker.MarkDirty;
             PoolManager.i.OnGet += Environment.i.platform.cullingController.objectsTracker.MarkDirty;
+
+            DCL.Environment.i.platform.updateEventHandler.AddListener(IUpdateEventHandler.EventType.Update, Update);
```

Comment:
it would be great to inject all needed dependencies from constructor instead of nesting singletons. It makes it harder to test and/or scale adding implicit coupling. This class is really big which would take probably a big refactor, out of the scope of this PR. Technical debt?

Reply:
Removing responsibilities from `SceneController` is an ongoing effort that goes back to the start of the project! We'll continue doing that for sure. About nesting singletons there's a justification but's too wally of texty to convey here, to summarize, the design of this will be greatly improved when the dcl runtime is moved to its own plugin.